### PR TITLE
add Batch Withdraw

### PR DESCRIPTION
**Pull request by Lomet**

### CODE REVIEW

File: contracts/util/BatchWithdraw.sol [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import "../interfaces/ILockDealNFT.sol";
+
+contract BatchWithdraw {
+    ILockDealNFT public lockDealNFT;
+
+    constructor(ILockDealNFT _lockDealNFT) {
+        lockDealNFT = _lockDealNFT;
+    }
+
+    function batchWithdraw(uint256[] memory tokenIds) external {
```
Comment (YouStillAlive) [inline contracts/util/BatchWithdraw.sol:13]: better to use **calldata** instead of **memory**

File: contracts/util/BatchWithdraw.sol [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        bool isApproved = lockDealNFT.isApprovedForAll(msg.sender, address(this));
+        for (uint256 i = 0; i < tokenIds.length; i++) {
```
Comment (YouStillAlive) [inline contracts/util/BatchWithdraw.sol:15]: `++i` beats `i++` in gas tests

Comment (YouStillAlive) [inline contracts/util/BatchWithdraw.sol]: if there are a lot of nfts, it's better to create a local variable for `tokenIds.length`

**Review by ashwinarora — APPROVED** [review]
Looks good!

The `--i` can be wrapped inside an `unchecked` block to save gas.

All the `internal` functions can be marked `private` if there won't be a child contract.

It would be nice to have 2 new external view functions, `getTotalWithdrawableOfToken` and `getTotalRefundAmount`.

**Comment by YouStillAlive** [comment]
`BatchWithdraw` has been moved to a separate repository: https://github.com/The-Poolz/BatchWithdraw/

**Pull request closed without merging**